build_deploy.sh

```diff
@@ -27,7 +27,18 @@ git clone --branch $AUTOTUNE_GIT_BRANCH https://github.com/kruize/autotune.git $
 pushd $AUTOTUNE_GIT_DIR
 IMAGE_TAG=$(git rev-parse --short=7 HEAD)
 
-DOCKER_CONF="$PWD/.docker"
+# Create tmp dir to store data in during job run (do NOT store in $WORKSPACE)
+export TMP_JOB_DIR=$(mktemp -d -p "$HOME" -t "jenkins-${JOB_NAME}-${BUILD_NUMBER}-XXXXXX")
+echo "job tmp dir location: $TMP_JOB_DIR"
+
+function job_cleanup() {
+    echo "cleaning up job tmp dir: $TMP_JOB_DIR"
+    rm -fr $TMP_JOB_DIR
+}
+
+trap job_cleanup EXIT ERR SIGINT SIGTERM
+
+DOCKER_CONF="$TMP_JOB_DIR/.docker"
 mkdir -p "$DOCKER_CONF"
 docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
 docker --config="$DOCKER_CONF" login -u="$RH_REGISTRY_USER" -p="$RH_REGISTRY_TOKEN" registry.redhat.io
```

Comment:
2 comments here:
- You can leverage the variable `DOCKER_CONFIG` (from [docker documentation](https://docs.docker.com/engine/reference/commandline/cli/#change-the-docker-directory)) to avoid needing to specify `--config` parameter
- You should consider not using the `-p` to provide the password, and instead use STDIN instead. From [the docs](https://docs.docker.com/reference/cli/docker/login/#password-stdin)

```suggestion
DOCKER_CONFIG="$TMP_JOB_DIR/.docker"
mkdir -p "$DOCKER_CONF"
docker login -u="$QUAY_USER" --password-stdin quay.io <<< "$QUAY_TOKEN"
docker login -u="$RH_REGISTRY_USER" --password-stdin registry.redhat.io <<< "$RH_REGISTRY_TOKEN"
```

I can't suggest on further lines in the PR in this comment but you can feel free to remove the `--config` parameter for every other command.

Reply:
I added the `export` to `DOCKER_CONFIG` otherwise it won't work

Reply:
Done